zookeeper-server/src/main/java/org/apache/zookeeper/cli/RestoreCommand.java

```diff
@@ -171,6 +181,26 @@ public boolean exec() throws CliException {
       System.out.println(getUsageStr());
       return true;
     }
+    if (cl.hasOption(OptionShortForm.OVERWRITE)) {
+      Scanner scanner = new Scanner(System.in);
+      boolean repeat = true;
+      while (repeat) {
+        System.out.println(
+            "Are you sure you want to overwrite the destination directories? Please enter \"yes/no\".");
+        String input = scanner.nextLine().toLowerCase();
+        switch (input) {
+          case "yes":
+            repeat = false;
+            break;
+          case "no":
+            System.out.println(
+                "Exiting restoration. Please remove the \"overwrite\" option from the command, and try again.");
+            return true;
+          default:
+            break;
```

Comment:
NIT: let's not silently let it mark the repeat as true. Because if people typo it as "yess" then it will falls to default. So either we fail it, or let it to "true", we need let user know about it.

Reply:
Good suggestion. I added a message to let the user know that their input cannot be recognized, meanwhile repeat prompting the message to user and ask them to type in "yes" or "no".